src/test/dependencies.test.ts

```diff
@@ -65,6 +68,47 @@ suite("dependencies", () => {
             listener => myDependency.install("remote", true, listener));
     });
 
+    test("Get tagged asset url", async function() {
+        this.timeout(PRUSTI_UBUNTU_ASSET_DOWNLOAD_TIMEOUT_MS);
+        const assetName = "prusti-release-ubuntu.zip";
+        const url = await GitHubReleaseAsset.getTaggedAssetUrl(
+            "viperproject", "prusti-dev", assetName, "v-2020-10-06-1807");
+        console.info(`URL to prusti-dev Ubuntu asset of v-2020-10-06-1807 release: '${url}'`);
+
+        // now download the zip:
+        const headers = {
+            "Accept": "application/octet-stream"
+        };
+        const myDependency = new Dependency<"remote">(
+            TMP_PATH,
+            ["remote",
+                new InstallerSequence([
+                    new FileDownloader(url, headers, assetName)
+                ])
+            ]
+        );
+        const { result: downloadDestination } = await withProgressInWindow(
+            "Testing download of a GitHub asset",
+            listener => myDependency.install("remote", true, listener));
+        console.log(`Prusti asset downloaded to '${downloadDestination.basePath}'`);
+        // check md5 of this file
+        const expected: string = "f7d4dd24eb40c0375d7a37d839ca515b";
+        const actual: string = await md5File(downloadDestination.basePath);
+        assert.strictEqual(actual, expected, `md5 hash of downloaded Prusti release asset does not match`);
+    });
+
+    test("Get latest pre-release asset url", async function() {
+        const url = await GitHubReleaseAsset.getLatestAssetUrl(
+            "viperproject", "prusti-dev", "prusti-release-ubuntu.zip", /* includePrerelease */ true);
+        console.info(`URL to latest non-pre- or pre-release prusti-dev Ubuntu asset is: '${url}'`);
+    });
+
+    test("Get latest asset url", async function() {
+        const url = await GitHubReleaseAsset.getLatestAssetUrl(
+            "viperproject", "prusti-dev", "prusti-release-ubuntu.zip");
+        console.info(`URL to latest non-pre-release prusti-dev Ubuntu asset is: '${url}'`);
+    });
```

Comment:
I would prefer creating an empty "immutable" repository (even outside our organization) with a bunch of releases and pre-releases, such that we can actually compare `url` with an expected url.

Reply:
I was thinking about comparing the received `url` but did not add a check for that as imho we do not have any guarantees from GitHub about what url will be returned. That was the reason why I've added a download of a release asset and check the asset's hash

Reply:
I see, good point. To keep the test short we could download the asset (e.g. with the `got` package) and compute the checksum without going through `FileDownloader`.

Reply:
I'll keep `FileDownloader` to simultaneously test it as well. I've now created a helper function to keep the individual test cases concise